src/Sylius/Behat/Context/Setup/CheckoutContext.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Bundle\ApiBundle\Command\Checkout\ChoosePaymentMethod;
+use Sylius\Bundle\ApiBundle\Command\Checkout\ChooseShippingMethod;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\BrowserKit\AbstractBrowser;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class CheckoutContext implements Context
+{
+    /** @var AbstractBrowser */
+    private $client;
+
+    /** @var RepositoryInterface */
+    private $shippingMethodRepository;
+
+    /** @var RepositoryInterface */
+    private $paymentMethodRepository;
+
+    /** @var MessageBusInterface */
+    private $commandBus;
+
+    /** @var SharedStorageInterface */
+    private $sharedStorage;
+
+    public function __construct(
+        AbstractBrowser $client,
+        RepositoryInterface $shippingMethodRepository,
+        RepositoryInterface $paymentMethodRepository,
+        MessageBusInterface $commandBus,
+        SharedStorageInterface $sharedStorage
+    ) {
+        $this->client = $client;
+        $this->shippingMethodRepository = $shippingMethodRepository;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+        $this->commandBus = $commandBus;
+        $this->sharedStorage = $sharedStorage;
+    }
+
+    /**
+     * @Given I have proceeded through checkout process
+     */
+    public function iHaveProceededThroughCheckoutProcess(): void
+    {
+        $this->addressOrder([
+            'email' => 'rich@sylius.com',
+            'billingAddress' => [
+                'city' => 'New York',
+                'street' => 'Wall Street',
+                'postcode' => '00-001',
+                'countryCode' => 'US',
+                'firstName' => 'Richy',
+                'lastName' => 'Rich',
+            ],
+        ]);
+
+        $command = new ChooseShippingMethod(0, $this->shippingMethodRepository->findOneBy([])->getCode());
+        $command->setOrderTokenValue($this->sharedStorage->get('cart_token'));
+
+        $this->commandBus->dispatch($command);
+
+        $command = new ChoosePaymentMethod(0, $this->paymentMethodRepository->findOneBy([])->getCode());
+        $command->setOrderTokenValue($this->sharedStorage->get('cart_token'));
+
+        $this->commandBus->dispatch($command);
+    }
+
+    private function addressOrder(array $content): void
+    {
+        if (!array_key_exists('email', $content)) {
+            $content['email'] = null;
+        }
+
+        $this->client->request(
```

Comment:
You should dispatch commands instead of calling an API endpoint in setup context